api/v1beta1/tags.go

```diff
@@ -95,6 +98,27 @@ func (t Tags) Validate() []*field.Error {
 				field.Invalid(field.NewPath("spec", "additionalTags"), v, "value cannot be longer than 256 characters"),
 			)
 		}
+		if len(k) > 3 && k[0:4] != "aws:" {
```

Comment:
Can we add unit tests for the new validations, so as to check that the logic works fine?

Reply:
Can we refactor this code to a func, something like:
```go
func isUserOwnedTag() bool {
   return len(k) > 3 && k[0:4] != "aws:" 
}
```
so as to improve code readability?